compiler/src/dotty/tools/dotc/transform/localopt/StringContextChecker.scala

```diff
@@ -1,13 +1,20 @@
-// ALWAYS KEEP THIS FILE IN src-bootstrapped, DO NOT MOVE TO src
-
-package dotty.internal
-
-import scala.quoted._
-
-object StringContextMacro {
-
-  /** Implementation of scala.StringContext.f used in Dotty */
-  inline def f(inline sc: StringContext)(inline args: Any*): String = ${ interpolate('sc, 'args) }
+package dotty.tools.dotc
+package transform.localopt
+
+import dotty.tools.dotc.ast.Trees._
+import dotty.tools.dotc.ast.tpd
+import dotty.tools.dotc.core.Decorators._
+import dotty.tools.dotc.core.Constants.Constant
+import dotty.tools.dotc.core.Contexts._
+import dotty.tools.dotc.core.StdNames._
+import dotty.tools.dotc.core.NameKinds._
+import dotty.tools.dotc.core.Symbols._
+import dotty.tools.dotc.core.Types._
+
+// Ported from old dotty.internal.StringContextMacro
+// TODO: port Scala 2 logic? (see https://github.com/scala/scala/blob/2.13.x/src/compiler/scala/tools/reflect/FormatInterpolator.scala#L74)
```

Comment:
What does the Scala 2 logic do that is different?

Reply:
It is a completely different implementation. Not sure which one is better. Someone will have to check that at some point.

Reply:
scala 2 `f` goes overboard to verify the format, but is it worthwhile maintaining code for a slow string mechanism.

Reply:
As far as I remember this implementation covered all the Scala 2 `f` interpolator test cases. They diverged slightly on the second error after some kinds of error in the format part of the string.

Reply:
We might have missed something. It would be ok for me to use the Scala 2 implementation but it would take a more effort to port. I could have a look in the future or let someone else try it. The current priority is to unblock #9925.